### PR TITLE
LTS: Implement Go version upgrade and switch to GitHub Actions

### PR DESCRIPTION
**Pull request by roidelapluie**
This commit upgrades the underlying Go version from 1.18 to 1.19 in order to align with the Prometheus main branch and ensure that the Long-Term Support (LTS) release remains under a supported version of Go. The upgrade also requires changes to the build process for Windows. Additionally, this commit switches the CI/CD pipeline from CircleCI to GitHub Actions for better integration and alignment with the Prometheus main branch.

This is following an email sent to the prometheus-developers mailing list in december.

Signed-off-by: Julien Pivotto <roidelapluie@o11y.eu>

### CODE REVIEW

File: .github/workflows/buf-lint.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 name: buf.build
-on:
+on:  # yamllint disable-line rule:truthy
```
Comment (mmorel-35) [inline .github/workflows/buf-lint.yml:2]: Hi @roidelapluie ,
I created #11871 so you dont have to add this comment on each github workflow file

Comment (roidelapluie) [reply .github/workflows/buf-lint.yml]: thanks, however this branch is just mirroring main so it's not relevant for this specific pr.

**Comment by roidelapluie** [comment]
cc @juliusv @SuperQ

**Comment by juliusv** [comment]
:+1:

**Review by SuperQ — APPROVED** [review]

**Pull request merged**